### PR TITLE
getUniqueValues getter for Phase 2 Store Refactor

### LINKED ISSUE #277: create getter for unique values of a category
This getter, `getUniqueValues` takes a category (and an optional number to limit number of unique values returned) and compiles a list of all unique values in the columns linked to that category – as long as they are *not missing* values.

- [x] depending components have been checked and are calling this mutation correctly
- [x] have unit tests to check functionality
- [x] needed store state variables exist already or have been created if not

### PR DESCRIPTION
**Pull request by jarmoza**
This PR closes #277 by implementing the following:

1. `getUniqueValues` getter in the store that takes every column linked to a given category and gathers unique values for each column via the `dataTable`. The getter outputs an object keyed on column with values that are an array of the unique values for that column. Missing values are skipped via lookup in the `dataDictionary`
2. `getUniqueValues` also takes an optional parameter `maxValues` to limit how many values are returned
3. `displayItems` in the `annot-categorical` component has been rewritten to accommodate the new output of `getUniqueValues`
4. Unit tests for this getter in `store-getter-getUniqueValues.cy.js` test the following scenarios: a) unique values for a single column, b) unique values for multiple columns, c) retrieving unique values with the optional maximum amount argument, and d) ensuring that no 'missing' values are returned with the unique values

### CODE REVIEW

File: store/index.js [hunk 1]
```diff
@@ -240,6 +241,44 @@ export const getters = {
         return p_state.transformationHeuristics[columnDataType];
     },
 
+    getUniqueValues: (p_state) => (p_category, p_maxValues="None") => {
+
+        // 1. Construct a list of unique values for each column
```
Comment (rmanaem) [inline store/index.js:246]: I think this should be `Construct an object containing...` since `uniqueValues` is an object

File: store/index.js [hunk 2]
```diff
@@ -240,6 +241,44 @@ export const getters = {
...
+                // NOTE: Trimming is done here instead of only looking at p_maxValues rows
+                // just in case there are blank entries for columns in the data table
+                if ( "None" !== p_maxValues ) {
+
+                    uniqueValues[columnName] = uniqueValues[columnName].slice(0, p_maxValues);
+                }
+            }
+        }
+
+        // Return a list of objects
```
Comment (rmanaem) [inline store/index.js:278]: I think this should be `Returns an object ...`

**Review by rmanaem — APPROVED** [review]
Great work on the PR @jarmoza! The unit-test test is thorough and in-line comments are nice and helpful. I left two comments that simply address the type of `uniqueValues` mentioned in the in-line comments other than that  It's good to go!

File: store/index.js (hunk 1 above)
Comment (jarmoza) [reply store/index.js:246]: Good point. Comment needed to be updated with implementation change.

**Pull request merged**